### ISSUE TITLE
Request to Make Options to pass SHA/Ref

### ISSUE THREAD

**Issue by Emily-Thomas** [jaywcjlove/github-action-modify-file-content#2]
Is it possible to have this extended to pass in a SHA or Ref instead of defaulting to the default branch? I would like to use this to make a commit not to my master, but to a designated branch so that I can generate a PR into my master branch. 

Please suggest if this is already possible.

**Comment by jaywcjlove** [comment]
I added `ref` input parameter.
@Emily-Thomas

**Comment by Emily-Thomas** [comment]
Thank you so much, I will test this out and let you know.

**Comment by Emily-Thomas** [comment]
@jaywcjlove I am trying to use this but am getting the error `[PATH] does not match [REF]` after modifyPathContents Body: runs

**Comment by Emily-Thomas** [comment]
Here is a copy of my setup. 

`
    - name: New Branch
      uses: peterjgrainger/action-create-branch@v2.2.0
      env:
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      with:
        branch: 'bossaccess-${{ steps.date.outputs.date }}'
        sha: '${{ github.event.pull_request.head.sha }}'
    - uses: actions/checkout@v3
      with:
        ref: 'bossaccess-${{ steps.date.outputs.date }}'
    - name: Modify Conf
      uses: jaywcjlove/github-action-modify-file-content@main
      with:
        path: conf/environments/prod/boss.conf
        body: ${{ steps.getdata.outputs.data }}
        ref: 'bossaccess-${{ steps.date.outputs.date }}'
`

**Comment by Emily-Thomas** [comment]
If I had to guess the error is getting thrown here https://github.com/jaywcjlove/github-action-modify-file-content/blob/cf1ca3c2d93367421f510349ae2e75b2ddb46cb8/src/index.ts#L44

**Comment by jaywcjlove** [comment]
@Emily-Thomas 

### specify branch changes

```yml
- name: Modify test test/overwrite.file.md
  uses: github-action-modify-file-content@main
  with:
    branch: test
    path: test/overwrite.file.md
    body: "{{date:YYYY-MM-DD HH:mm:ss}}"
    overwrite: 'true'
```

https://github.com/jaywcjlove/github-action-modify-file-content/blob/58e4b5581d8e3b61025c4e8b09888ae516a256aa/.github/workflows/ci.yml#L28-L35

**Comment by Emily-Thomas** [comment]
Okay, I updated to use that format. Now the logging is correct in the action workflow, but the commit that is made is an empty commit. I've tried a few combinations of using path + ref, only path, only ref, using overwrite/not using overwrite and the result is the same, an empty commit. However when I go through the logging of the function, it is grabbing the right tag in the file and replacing it - just not saving or commiting those changes. @jaywcjlove - I appreciate the timely help and changes here.
